crates/bevy_reflect/src/impls/std.rs

```diff
@@ -220,6 +220,171 @@ impl_type_path!(::bevy_utils::NoOpHash);
 impl_type_path!(::bevy_utils::EntityHash);
 impl_type_path!(::bevy_utils::FixedState);
 
+macro_rules! impl_reflect_for_atomic {
+    ($ty:ty, $ordering:ident) => {
+        const _: () = {
+            #[allow(unused_mut)]
+            impl bevy_reflect::GetTypeRegistration for $ty
+            where
+                $ty: ::core::any::Any + ::core::marker::Send + ::core::marker::Sync,
+            {
+                fn get_type_registration() -> bevy_reflect::TypeRegistration {
+                    let mut registration = bevy_reflect::TypeRegistration::of::<Self>();
+                    registration.insert::<bevy_reflect::ReflectFromPtr>(bevy_reflect::FromType::<
+                        Self,
+                    >::from_type());
+                    registration.insert::<bevy_reflect::ReflectFromReflect>(
+                        bevy_reflect::FromType::<Self>::from_type(),
+                    );
+                    registration
+                        .insert::<ReflectDefault>(bevy_reflect::FromType::<Self>::from_type());
+                    registration
```

Comment:
Do the atomic types also support serde traits? If so, we could also register `ReflectSerialize` and `ReflectDeserialize`.